### PR TITLE
Containerize `./hack/tunnel`

### PR DESCRIPTION
**Pull request by tsatam**
### Which issue this PR addresses:

Fixes [ARO-8549](https://issues.redhat.com/browse/ARO-8549)

### What this PR does / why we need it:

Builds `./hack/tunnel` into a runnable container. This will allow it to eventually be used within CI contexts to make requests using dev tooling (e.g. `az aro` extension) to a running full-service RP. 

### Test plan for issue:

- [X] Manually ran tunnel container and used it to forward traffic to a running full-service RP instance. Example Podman invocation:
```sh
for f in "localhost.crt" "localhost.key" "dev-client.crt" "dev-client.key"; do
    podman secret create $f ./secrets/$f;
done

podman run -it \
    -p 8443:8443 \
    --secret localhost.crt,target=/opt/secrets/localhost.crt \
    --secret localhost.key,target=/opt/secrets/localhost.key \
    --secret dev-client.crt,target=/opt/secrets/dev-client.crt \
    --secret dev-client.key,target=/opt/secrets/dev-client.key \
    aro-tunnel:${VERSION} \
    --certFile /opt/secrets/localhost.crt \
    --keyFile /opt/secrets/localhost.key \
    --clientCertFile /opt/secrets/dev-client.crt \
    --clientKeyFile /opt/secrets/dev-client.key \
    $(az network public-ip show -g ${RESOURCEGROUP} -n rp-pip --query 'ipAddress')
```

I have validated the above on Linux. I'm looking for validation on a macOS dev environment as well in order to ensure it is compatible. 

### Is there any documentation that needs to be updated for this PR?

No

### How do you know this will function as expected in production? 

Not a production change. This will be utilized in future CI efforts.

### CODE REVIEW

**Review by bitoku — APPROVED** [review]

**Review by SudoBrendan — APPROVED** [review]
Seems OK for linux - things get weird for stuff like this on MacOS (because "localhost" is podman machine), so there may be issues with things like your secrets and ports. In addition, is a standing VPN connection required to the VNet hosting the RP? I'd have some approvals from MacOS users before merge so we can get that sorted.

**Comment by tsatam** [comment]
> Seems OK for linux - things get weird for stuff like this on MacOS (because "localhost" is podman machine), so there may be issues with things like your secrets and ports. In addition, is a standing VPN connection required to the VNet hosting the RP? I'd have some approvals from MacOS users before merge so we can get that sorted.

A VPN connection is not required for this to work, the full-service RP instance is accessible publicly (using the certs utilized by this tunnel). 

Note that I do not convert the existing `make tunnel` target in this PR to use the tunnel image yet - my focus was on the CI use case (where this container will run within a k8s pod as a sidecar to e.g. the Azure CLI w/ ARO dev extension). 

I hope that this implementation would work for local dev and on macOS (I intentionally utilized podman secrets to share the necessary cert/keys rather than volumes, and port auto-forwarding _should_ work), I cannot validate this as I do not run macOS.

**Review by jaitaiwan — APPROVED** [review]

**Comment by jaitaiwan** [comment]
I dont think we have any macOS users in APAC.

**Pull request merged**
